airflow/utils/db.py

```diff
@@ -92,7 +92,7 @@ def _format_airflow_moved_table_name(source_table, version, category):
 @provide_session
 def merge_conn(conn, session: Session = NEW_SESSION):
     """Add new Connection."""
-    if not session.query(conn.__class__).filter_by(conn_id=conn.conn_id).first():
+    if not session.scalar(select(conn.__class__).filter_by(conn_id=conn.conn_id).limit(1)):
```

Comment:
Not strictly related but we should refactor this to use `count()` or `SELECT TRUE` to avoid populating the Python object unnecessarily.

Reply:
This PR just replaces the `session.query`. @ephraimbuddy and I were planning to do the re-writing in another PR.

Reply:
Isn't this `.exists()`? https://docs.sqlalchemy.org/en/20/orm/queryguide/query.html#sqlalchemy.orm.Query.exists

Reply:
That’s the old API though, not sure if an equivalent exists in the new API. If not we’ll need to roll our own (there are several choices)